roachtest: npgsql failed

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14417897?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14417897?buildTab=artifacts#/npgsql) on master @ [34caf9d2c267f95f6aef2713708806e6f14948d6](https://github.com/cockroachdb/cockroach/commits/34caf9d2c267f95f6aef2713708806e6f14948d6):


```
(orm_helpers.go:221).summarizeFailed: 
Tests run on Cockroach v24.1.0-alpha.3-dev-34caf9d2c267f95f6aef2713708806e6f14948d6
Tests run against npgsql v7.0.2
3321 Total Tests Run
2675 tests passed
646 tests failed
268 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: Npgsql.Tests.CopyTests(Multiplexing).Prepended_messages -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14417897/npgsql/1710590288011/1710590590691)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119240 roachtest: npgsql failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-36767

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14453099?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14453099?buildTab=artifacts#/npgsql) on master @ [a2f1f379ee52ceee2b2aa6769f1fa162f8d6b8a7](https://github.com/cockroachdb/cockroach/commits/a2f1f379ee52ceee2b2aa6769f1fa162f8d6b8a7):


```
(orm_helpers.go:221).summarizeFailed: 
Tests run on Cockroach v24.1.0-alpha.4-dev-a2f1f379ee52ceee2b2aa6769f1fa162f8d6b8a7
Tests run against npgsql v7.0.2
3322 Total Tests Run
2677 tests passed
645 tests failed
267 tests skipped
0 tests ignored
0 tests passed unexpectedly
2 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: Npgsql.Tests.CopyTests(Multiplexing).Cancel_binary_import -  (unexpected)
--- FAIL: Npgsql.Tests.CopyTests(Multiplexing).Import_commit_in_middle_of_row -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14453099/npgsql/1710838859464/1710839165128)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #120606 roachtest: npgsql failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/cmd/roachtest/tests/dotnet_helpers.go
+++ b/pkg/cmd/roachtest/tests/dotnet_helpers.go
@@ -83,6 +83,7 @@ func extractFailureFromTRX(contents []byte) ([]string, []status, map[string]stri
 
 	npgsqlFlakeErrors := []string{
 		"Received backend message ReadyForQuery while expecting",
+		"Received unexpected backend message ReadyForQuery",
 		"Got idle connector but State is Copy",
 	}
 	isAnyFlakeError := func(s string) bool {

